core/node/node_framework/src/implementations/layers/eth_watch.rs

```diff
@@ -66,21 +73,54 @@ impl WiringLayer for EthWatchLayer {
         let main_pool = input.master_pool.get().await?;
         let client = input.eth_client.0;
 
-        let eth_client = EthHttpQueryClient::new(
+        let sl_diamond_proxy_addr = if self.settlement_mode.is_gateway() {
+            self.gateway_contracts_config
+                .clone()
+                .unwrap()
```

Comment:
Is `unwrap()` safe here by construction? Prefer propagating errors if there's even a remote chance that it can be triggered.

Reply:
I think it can be triggered, it is more like invalid configuration state, I will propagate